### PR TITLE
Ship API

### PR DESCRIPTION
**Pull request by ewoudje**
As spoken of in discord, this will replace AttachedStorage and offer a more generic API

So interface Ship is the representation of just a generic ship, to be used when you don't care when its loaded or not.
But if you know its loaded or not loaded please use the correct class!

The interface ShipProvider is just a generic interface that makes it handier for functions to pass objects who aren't ships but have the ship data. This is very handy for any generic ship function.

A couple of ideas:
 * Renaming ShipData and ShipObject and all their children 
 * An API for the ship world and while were at it to make it also dimension independent

### CODE REVIEW

**Review by StewStrong — COMMENTED** [review]
Looks good to me! But maybe we should see what Ruby thinks before merging.

File: src/main/kotlin/org/valkyrienskies/core/api/Ship.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package org.valkyrienskies.core.api
+
+import org.joml.Matrix4dc
+
+interface Ship : ShipProvider {
+    val shipyardToWorld: Matrix4dc
```
Comment (Rubydesic) [inline src/main/kotlin/org/valkyrienskies/core/api/Ship.kt:6]: super nitpicky but shipToWorld and worldToShip are shorter and convey the same info imo

**Review by Rubydesic — APPROVED** [review]

**Comment by ewoudje** [comment]
So the interface is now implemented for ShipData and ShipObjectServer; this means this PR should be done. Merge if good

File: src/test/kotlin/org/valkyrienskies/core/api/ShipApiTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package org.valkyrienskies.core.api
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import org.valkyrienskies.core.VSRandomUtils
+import org.valkyrienskies.core.game.ships.ShipObjectServer
+
+// Yes its a very simple test, but if somebody ever dares to break it we will know
```
Comment (StewStrong) [inline src/test/kotlin/org/valkyrienskies/core/api/ShipApiTest.kt:8]: 😂

**Comment by StewStrong** [comment]
Freaking sweat!

**Pull request merged**